internal/core/src/segcore/ConcurrentVector.h

```diff
@@ -229,6 +230,27 @@ class ConcurrentVectorImpl : public VectorBase {
         return chunks_[chunk_index].data();
     }
 
+    // Heavy and expensive, keep this not frequently.
+    // This function should only be called when small index will be created.
+    // An index chunk corresponds to many data chunks, so multiple data chunks will be assembled into an index chunk.
+    Chunk
+    get_chunks(const std::vector<ssize_t>& chunk_ids) const {
+        std::vector<ssize_t> prefix_sums(1, 0);
+        auto n = chunk_ids.size();
+        for (int i = 0; i < n; i++) {
+            auto chunk_id = chunk_ids[i];
+            auto& chunk = get_chunk(chunk_id);
+            prefix_sums.push_back(prefix_sums[i] + chunk.size());
```

Comment:
define prefix_sums with correct size to avoid resizing:
std::vector<ssize_t> prefix_sum(chunk_ids.size() + 1, 0);